library/src/main/java/com/google/android/cameraview/CameraView.java

```diff
@@ -407,13 +411,16 @@ public void takePicture() {
         mImpl.takePicture();
     }
 
-    private class CallbackBridge implements CameraViewImpl.Callback {
+    private static class CallbackBridge implements CameraViewImpl.Callback {
```

Comment:
My team uses LeakCanary to detect memory leaks.  Unfortunately, the warnings were not false-positives.  When I implemented these simple changes, the warnings went away.
It might be helpful to integrate LeakCanary into the demo app.  It only takes a few minutes and run the app a few times to find out the leaks.